src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.cpp

```diff
@@ -90,43 +119,141 @@ void Instance::SetCookTime(uint32_t cookTime)
     }
 }
 
-void Instance::SetPowerSetting(uint8_t powerSetting)
-{
-    uint8_t oldPowerSetting = mPowerSetting;
-    mPowerSetting           = powerSetting;
-    if (mPowerSetting != oldPowerSetting)
-    {
-        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::PowerSetting::Id);
-    }
-}
-
 CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
 {
-    ChipLogError(Zcl, "OperationalState: Reading");
+    ChipLogError(Zcl, "Microwave Oven Control: Reading");
     switch (aPath.mAttributeId)
     {
     case MicrowaveOvenControl::Attributes::CookTime::Id: {
         ReturnErrorOnFailure(aEncoder.Encode(GetCookTime()));
     }
     break;
 
+    case MicrowaveOvenControl::Attributes::MaxCookTime::Id: {
+        ReturnErrorOnFailure(aEncoder.Encode(mDelegate->GetMaxCookTime()));
+    }
+    break;
+
     case MicrowaveOvenControl::Attributes::PowerSetting::Id: {
-        ReturnErrorOnFailure(aEncoder.Encode(GetPowerSetting()));
+        if (HasFeature(MicrowaveOvenControl::Feature::kPowerAsNumber))
+        {
+            ReturnErrorOnFailure(aEncoder.Encode(mDelegate->GetPowerSetting()));
+        }
+        else
+        {
+            ChipLogError(Zcl, "Microwave Oven Control: feature is not supported");
```

Comment:
I think reading the logs would be easier if we include the feature name in the log text, otherwise we just know this is some feature of MOC. How about "Microwave Oven Control: cannot get power setting number: PowerAsNumber feature is not supported" or similar?